index.js

```diff
@@ -0,0 +1,167 @@
+var instance_skel  = require('../../instance_skel');
+const WebSocket    = require('ws');
+var objectPath     = require('object-path');
+
+class instance extends instance_skel {
+	constructor(system, id, config) {
+		super(system, id, config);
+
+		this.ws = null;
+		this.subscriptions = new Map();
+
+		this.actions();
+		this.initFeedbacks();
+		this.subscribeFeedbacks()
+
+		if (!this.config) {
+			return this;
+		}
+
+		this.initWebSocket();
+
+		return this;
+	}
+
+	updateConfig(config) {
+		this.config = config;
+		this.setVariableDefinitions([{}]);
+		this.initWebSocket();
+	}
+
+	initWebSocket() {
+		var ip = this.config.host;
+		var port = this.config.port;
+		this.status(this.STATUS_UNKNOWN)
+		if (!ip || !port) {
+			this.status(this.STATUS_ERROR)
+			return;
+		}
+		this.ws = new WebSocket(`ws://${ip}:${port}`);
+
+		this.ws.on('open', () => {
+			this.log('debug', `Connection opened`);
+			this.status(this.STATUS_OK)
+		});
+		this.ws.on('close', () => {
+			this.log('debug', `Connection closed`);
+			this.status(this.STATUS_ERROR)
+		});
+
+		this.ws.on('message', (data) => {
+			this.messageReceivedFromWebSocket(data);
+		});
+
+		this.ws.on('error', (data) => {
+			this.log('error', `WebSocket error: ${data}`);
+		});
+	}
+
+	messageReceivedFromWebSocket(data) {
+		this.log('debug', `Message received: ${data}`);
```

Comment:
be aware that this can become spammy ... for a generic module it can make sense, but for general module dev this would need to be removed.

Reply:
I have added debug configuration switch to enable message (in/out) logging.